src/Compiler/Checking/AttributeChecking.fs

```diff
@@ -236,8 +238,20 @@ let private CheckILAttributes (g: TcGlobals) isByrefLikeTyconRef cattrs m =
     | Some ([ILAttribElem.String (Some msg) ], _) when not isByrefLikeTyconRef -> 
             WarnD(ObsoleteWarning(msg, m))
     | Some ([ILAttribElem.String (Some msg); ILAttribElem.Bool isError ], _) when not isByrefLikeTyconRef -> 
-        if isError then 
-            ErrorD (ObsoleteError(msg, m))
+        if isError then
+            if g.langVersion.SupportsFeature(LanguageFeature.RequiredPropertiesSupport) then
+                // In some cases C# will generate both ObsoleteAttribute and CompilerFeatureRequiredAttribute.
+                // Specifically, when default constructor is generated for class with any reqired members in them.
+                // ObsoleteAttribute should be ignored if CompilerFeatureRequiredAttribute is present, and its name is "RequiredMembers".
+                // REVIEW: Shall feature names be moved to LanguageFeatures (or elsewhere), and be tied to actual features somehow?
+                let (AttribInfo(tref,_)) = g.attrib_CompilerFeatureRequiredAttribute
+                match TryDecodeILAttribute tref cattrs with
+                | Some([ILAttribElem.String (Some featureName) ], _) when featureName = "RequiredMembers" ->
```

Comment:
Not sure, if we need to have a some more abstract mechanism of checking specific features `CompilerFeatureRequiredAttribute`, maybe tied to language versions?

I think I will see how do we use `CompilerFeatureRequiredAttribute` with ref fields and decide if we need some abstract way of doing it.